pkgs/top-level/all-packages.nix

```diff
@@ -1004,6 +1004,36 @@ with pkgs;
     ];
   } ../build-support/setup-hooks/validate-pkg-config.sh;
 
+  autoPatchPcHook =
+    makeSetupHook
+      {
+        name = "auto-patch-pc-hook";
+        propagatedBuildInputs = [ pkg-config ];
+      }
+      (
+        writeScript "patch-pc.sh" ''
+          fixupOutputHooks+=(autoPatchPcHook)
+
+          autoPatchPcHook() (
+              # Some packages list their own libraries as deps, they should be searched first in case of cyclic deps
+              export PKG_CONFIG_PATH_${pkgconf.suffixSalt}="$prefix/lib/pkgconfig:$prefix/share/pkgconfig:$PKG_CONFIG_PATH"
```

Comment:
I understand `PKG_CONFIG_PATH_<...>` won't work with `pkgconf` anymore. The dummy propagation does prepare for `pkgconf`, but does the rest then actually work?

Reply:
Alright then, disregard - i thought i read something like that, sorry